src/configuration/OpenSearchDetails.tsx

```diff
@@ -18,14 +20,72 @@ const indexPatternTypes = [
 type Props = {
   value: DataSourceSettings<OpenSearchOptions>;
   onChange: (value: DataSourceSettings<OpenSearchOptions>) => void;
+  saveOptions: () => Promise<void>;
+  datasource: OpenSearchDatasource;
 };
 export const OpenSearchDetails = (props: Props) => {
-  const { value, onChange } = props;
+  const { value, onChange, saveOptions, datasource } = props;
+  const [versionErr, setVersionErr] = useState<string>('');
+  const setVersion = async () => {
+    setVersionErr('');
+    await saveOptions();
+    try {
+      const version = await datasource.getOpenSearchVersion();
+      onChange({
+        ...value,
+        jsonData: {
+          ...value.jsonData,
+          version: version.version,
+          flavor: version.flavor,
+          maxConcurrentShardRequests: getMaxConcurrentShardRequestOrDefault(
+            version.flavor,
+            version.version,
+            value.jsonData.maxConcurrentShardRequests
+          ),
+        },
+      });
+    } catch (error) {
+      let message = String(error);
+      if (error instanceof Error) {
+        message = error.message;
+      }
+      setVersionErr(message);
+    }
+  };
+
+  let versionString = '';
+  if (value.jsonData.flavor && value.jsonData.version) {
+    versionString = `${AVAILABLE_FLAVORS.find(f => f.value === value.jsonData.flavor)?.label ||
+      value.jsonData.flavor} ${value.jsonData.version}`;
+  }
+
+  const getServerlessSettings = (event: React.SyntheticEvent<HTMLInputElement, Event>) => {
```

Comment:
All of the query code assumes that a version and flavor are set, so rather than refactor that in this commit, I set the flavor and version when serverless is toggled. (I checked and you cannot get a version from serverless). I figured just setting it to the latest one was ok, as we'll need to go make other code changes anyway if a new latest version introduces breaking changes.

Reply:
Do we need a version/flavor if it's serverless? Aren't all serverless the same flavor/version?

Reply:
I tested this with a serverless datasource and the flow as a bit weird: 

https://user-images.githubusercontent.com/6620164/235996739-a9d00519-291c-4bd2-84c7-2001e46e5dc7.mov

what if when we get/set version we check if it's serverless based on the url and then set the toggle? 

I think we should be able to dynamically determine it based on the url, I think for example most of the endpoints are formatted like `https://{someidorsomething}.{region}.aoss.amazonaws.com`

I wonder if we could do a regex for `aoss.amazonaws.com` and if it's there we set the version to serverless or something?